File: web/src/api/article.ts
import request from './request'
import type { ArticleDetailResp } from '@/types'

// 文章详情
export function getArticleDetail(article_id: number) {
  return request.get<any, ArticleDetailResp>('/article/detail', { params: { article_id } })
}

// 保存文章草稿
// 后端 ArticleDraftReq 字段：title / body_md / cover_url / tags / images
// 后端返回 { article_id: number }
export function saveArticleDraft(data: {
  title: string
  body_md: string
  cover_url?: string
  tags?: string[]
  images?: string[]
}) {
  return request.post<any, { article_id: number }>('/article/draft', data)
}

// 发布文章
// 后端 ArticlePublishReq 字段：article_id（仅需 article_id 即可发布已有草稿）
export function publishArticle(article_id: number) {
  return request.post<any, void>('/article/publish', { article_id })
}

File: server/internal/dto/request/article.go
package request

// ArticleDraftReq 保存文章草稿请求
type ArticleDraftReq struct {
	Title    string   `json:"title" binding:"required,max=80"`
	BodyMD   string   `json:"body_md" binding:"required"`
	CoverURL string   `json:"cover_url" binding:"max=1024"`
	Tags     []string `json:"tags"`
	Images   []string `json:"images"`
}

// ArticlePublishReq 发布文章请求
type ArticlePublishReq struct {
	ArticleID uint64 `json:"article_id" binding:"required"`
}
